### PR TITLE
Teach photos service to do incremental updates

### PR DESCRIPTION
**Pull request by torarnv**
Fixes issue #85

### CODE REVIEW

**Comment by coddingtonbear** [comment]
Looks totally reasonable to me, @torarnv; I'm not super familiar with the underlying API, but if you've tested this and given it a shot, merge away (after maybe fixing the style nits that the test runner found).

Cheers!

**Comment by chrismetcalf** [comment]
Any updates on when this will be merged?

**Comment by torarnv** [comment]
Sorry guys, I'm swamped at work and with other projects at the moment 🙁 But I'll get to this as soon as I can, work out the kinks, and merge it.

In the meantime (for master, without this parch merged), doing something like `delattr(api, '_photos')` between each access to the photos service should force a refresh.

**Comment by bobhavlin** [comment]
Doing the `delattr(api, '_photos')` worked for me!
Thanks.

**Comment by torarnv** [comment]
I've limited this in scope to doing full updates of the service when updates are detected, instead of trying to granularly update things which is more fragile and is pointless unless we also expose an API for that.

Please test this PR guys, thanks!

File: README.rst [hunk 1]
```diff
@@ -270,3 +270,27 @@ To download a specific version of the photo asset, pass the version to ``downloa
 >>> download = photo.download('thumb')
 >>> with open(photo.versions['thumb'].filename, 'wb') as thumb_file:
         thumb_file.write(download.raw.read())
+
+To check if the photo library has been updated, call ``update()``:
+
+>>> api.photos.update()
+True
+
+Note: Updating does not invalidate PhotAlbum or PhotoAssets, so the albums and assets must be accessed again from the photos service:
```
Comment (torarnv) [inline README.rst]: Also, please note this part when testing.

**Comment by torarnv** [comment]
@picklepete a quick code review would be nice, ty! 😄

**Comment by Thamster** [comment]
>>> api.photos.update()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyicloud/base.py", line 289, in photos
    self.params
  File "pyicloud/services/photos.py", line 34, in __init__
    'syncToken': response['syncToken'],
KeyError: 'syncToken'

**Comment by torarnv** [comment]
@coddingtonbear hoping to merge this so that we can close #85 and cut a new release. That way we'll be able to track further issues with photos being stale (if that's still a problem), knowing that people at least are running with this patch in. What do you think?

**Comment by ndbroadbent** [comment]
I just wanted to mention that when I tried running this branch on a different iCloud account for the first time, I got this error:

```

**Comment by ndbroadbent** [comment]
Some other users have been running into this, so I ended up adding a note about this to [my project's README](https://github.com/ndbroadbent/icloud_photos_downloader#error-on-first-run). I'm also printing this output when you run my script for the first time:

```
Signing in...
Updating photos...
Bad Request (400)

This error usually means that Apple's servers are getting ready to send you data about your photos.
This process can take around 5-10 minutes, and it only happens when you run the script for the very first time.
Please wait a few minutes, then try again!

(If you are still seeing this message after 30 minutes, please open an issue on GitHub.)
```

I actually have no idea what's happening behind the scenes, so please let me know if that explanation is wrong. Am I correct that this is a problem with Apple's iCloud API, or is it a bug in the `#photos-update` branch?

**Comment by torarnv** [comment]
@ndbroadbent Thanks for the feedback, I'm guessing you're seeing a first-time startup corner case that I haven't observed yet since my account is not new. With a new account and some logging of first time Photos startup on icloud.com it should be possible to figure out what they are doing.

**Comment by ndbroadbent** [comment]
Hi @torarnv, sorry I should clarify, this is not with "new" iCloud account, but it keeps happening when you run the script on existing iCloud accounts for the first time. For example, I tried it on my wife's account (which is a few years old and has a few thousands photos), and I also got the `400 Bad Request` error on the first try. And then it went away after about 5 minutes.

**Comment by ndbroadbent** [comment]
Actually I think I do remember a "Preparing your library" message when I first signed into the iCloud website. Here's some other people that have the "Preparing" message: https://discussions.apple.com/thread/6886007?start=15&tstart=0

Mine was never stuck, but it seems like this a pretty common thing.

**Comment by torarnv** [comment]
Abandoned. The iCloud Photos API has changed.

**Pull request closed without merging**